ibm/service/vpc/resource_ibm_is_share_mount_target.go

```diff
@@ -0,0 +1,328 @@
+			},
+			"mount_target": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The unique identifier of this target",
+			},
+			"created_at": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The date and time that the share target was created.",
+			},
+			"href": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The URL for this share target.",
+			},
+			"lifecycle_state": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The lifecycle state of the mount target.",
+			},
+			"mount_path": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The mount path for the share.The IP addresses used in the mount path are currently within the IBM services IP range, but are expected to change to be within one of the VPC's subnets in the future.",
+			},
+			"resource_type": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The type of resource referenced.",
+			},
+		},
+	}
+}
+
+func ResourceIBMIsShareMountTargetValidator() *validate.ResourceValidator {
+	validateSchema := make([]validate.ValidateSchema, 1)
+	validateSchema = append(validateSchema,
+		validate.ValidateSchema{
+			Identifier:                 "name",
+			ValidateFunctionIdentifier: validate.ValidateRegexpLen,
+			Type:                       validate.TypeString,
+			Optional:                   true,
+			Regexp:                     `^([a-z]|[a-z][-a-z0-9]*[a-z0-9]|[0-9][-a-z0-9]*([a-z]|[-a-z][-a-z0-9]*[a-z0-9]))$`,
+			MinValueLength:             1,
+			MaxValueLength:             63,
+		},
+	)
+
+	resourceValidator := validate.ResourceValidator{ResourceName: "ibm_is_share_mount_target", Schema: validateSchema}
+	return &resourceValidator
+}
+
+func resourceIBMIsShareMountTargetCreate(context context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	vpcClient, err := meta.(conns.ClientSession).VpcV1BetaAPI()
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	createShareMountTargetOptions := &vpcbetav1.CreateShareMountTargetOptions{}
+
+	createShareMountTargetOptions.SetShareID(d.Get("share").(string))
+	vpcid := d.Get("vpc").(string)
+	vpc := &vpcbetav1.VPCIdentity{
+		ID: &vpcid,
+	}
+	createShareMountTargetOptions.SetVPC(vpc)
+	if _, ok := d.GetOk("name"); ok {
+		createShareMountTargetOptions.SetName(d.Get("name").(string))
+	}
+
+	shareTarget, response, err := vpcClient.CreateShareMountTargetWithContext(context, createShareMountTargetOptions)
+	if err != nil {
+		log.Printf("[DEBUG] CreateShareMountTargetWithContext failed %s\n%s", err, response)
+		return diag.FromErr(err)
+	}
+	_, err = WaitForTargetAvailable(context, vpcClient, *createShareMountTargetOptions.ShareID, *shareTarget.ID, d, d.Timeout(schema.TimeoutCreate))
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	d.SetId(fmt.Sprintf("%s/%s", *createShareMountTargetOptions.ShareID, *shareTarget.ID))
```

Comment:
Can we setID before waitingfor available. If soemtimes waiting intermittely the state file will hold ID and helps in future run

Reply:
moved above wait logic call